### PR TITLE
Fix/macro audit 8

### PR DESCRIPTION
**Pull request by crispymangoes**
This PR implements fixes described in the Macro Audit-8 Preliminary report.

Note M-1 and M-2 were fixed in #123 .

In addition to the fixes, the FTokenAdaptor was updated to replace the `ACCOUNT_FOR_INTEREST()` global function with an immutable bool. This makes deployments of FTokenAdaptors easier because we do not need to change the adaptor code, we only need to specify a different constructor input.

### CODE REVIEW

File: src/modules/adaptors/Frax/FTokenAdaptor.sol [hunk 1]
```diff
@@ -29,6 +29,17 @@ contract FTokenAdaptor is BaseAdaptor {
      */
     error FTokenAdaptor__FTokenPositionsMustBeTracked(address fToken);
 
+    /**
+     * @notice This bool determines how this adaptor accounts for interest.
```
Comment (0xEinCodes) [inline src/modules/adaptors/Frax/FTokenAdaptor.sol:33]: For pairs following `FraxlendPairCorev1` are we suggesting strategists to have this set as false, and then optional for `FraxlendPairCorev2`?

**Comment by kingofclubstroyDev** [comment]
Looks good to me

**Review by 0xEinCodes — CHANGES_REQUESTED** [review]
Just a couple of small changes and questions. Nice job!

File: src/modules/adaptors/Frax/FTokenAdaptor.sol [hunk 2]
```diff
@@ -29,6 +29,17 @@ contract FTokenAdaptor is BaseAdaptor {
...
+     *         True: Account for pending interest to be paid when calling `balanceOf` or `withdrawableFrom`.
+     *         False: Do not account for pending interest to be paid when calling `balanceOf` or `withdrawableFrom`.
+     */
+    bool public immutable ACCOUNT_FOR_INTEREST;
```
[the remark this answers is not in the document]
Comment (0xEinCodes) [reply src/modules/adaptors/Frax/FTokenAdaptor.sol:37]: From chatting with you on this it seems like my comment was on a different topic. We'll revisit all of this hopefully with a conversation with drake from Frax.

File: src/modules/adaptors/Morpho/MorphoAaveV2ATokenAdaptor.sol [hunk 3]
```diff
@@ -108,14 +101,15 @@ contract MorphoAaveV2ATokenAdaptor is BaseAdaptor, MorphoRewardHandler {
         // Make sure position is not backing a borrow.
         if (isBorrowingAny(address(this))) revert BaseAdaptor__UserWithdrawsNotAllowed();
 
-        IAaveToken aToken = abi.decode(adaptorData, (IAaveToken));
+        address aToken = abi.decode(adaptorData, (address));
```
[the remark this answers is not in the document]
Comment (0xEinCodes) [reply src/modules/adaptors/Morpho/MorphoAaveV2ATokenAdaptor.sol:104]: I will open up two issues that we can add stuff to until we get to refactoring and doc updates:

1. Issue 1 - Doc Updates
2. Issue 2 - Refactoring

**Review by 0xEinCodes — APPROVED** [review]
We discussed small comments and it all looks great.

**Pull request merged**